pkg/schedule/schedulers/scheduler_controller.go

```diff
@@ -470,29 +471,28 @@ func (s *ScheduleController) Schedule(diagnosable bool) []*operator.Operator {
 		if diagnosable {
 			s.diagnosticRecorder.SetResultFromPlans(ops, plans)
 		}
-		foundDisabled := false
+		if len(ops) == 0 {
+			continue
+		}
+
 		for _, op := range ops {
-			if labelMgr := s.cluster.GetRegionLabeler(); labelMgr != nil {
-				region := s.cluster.GetRegion(op.RegionID())
-				if region == nil {
-					continue
-				}
-				if labelMgr.ScheduleDisabled(region) {
-					denySchedulersByLabelerCounter.Inc()
-					foundDisabled = true
-					break
-				}
+			labelMgr := s.cluster.GetRegionLabeler()
+			if labelMgr == nil {
+				continue
 			}
-		}
-		if len(ops) > 0 {
-			// If we have schedule, reset interval to the minimal interval.
-			s.nextInterval = s.Scheduler.GetMinInterval()
-			// try regenerating operators
-			if foundDisabled {
+			region := s.cluster.GetRegion(op.RegionID())
+			if region == nil {
 				continue
 			}
-			return ops
+			if labelMgr.ScheduleDisabled(region) {
+				denySchedulersByLabelerCounter.Inc()
+				continue loop
```

Comment:
Did we forget to update `s.nextInterval = s.Scheduler.GetMinInterval()` when executing ops[0] successfully?

Reply:
L494

Reply:
`continue loop` will pass 494?

Reply:
fixed in https://github.com/tikv/pd/pull/8276/commits/f1c1a1eff5e5b9980736563da1046a808fdaf1e0

Seem the `Reset` does not work https://github.com/tikv/pd/blob/c015f140f49a47e60b3884a9179e79e92482461c/pkg/schedule/schedulers/scheduler_controller.go#L360-L361. 

`Interval` is fixed.

Reply:
Is it necessary to make this change?

Reply:
Did you mean the changes of this file or some others?

The changes of this file are:
1. decrease indent for code readability
2. use `continue loop` to instead with `foundDisabled` variable, this is more concise
3. no other updates